src/Engine/Physics/Trigger.cpp

```diff
@@ -0,0 +1,41 @@
+#include "Trigger.hpp"
+
+#include <btBulletDynamicsCommon.h>
+#include "RigidBody.hpp"
+#include "Shape.hpp"
+
+namespace Physics {
+
+    Trigger::Trigger(Shape* shape) : btCollisionWorld::ContactResultCallback() {
+        trigger = new btCollisionObject();
+        trigger->setCollisionShape(shape->GetShape());
+    }
+
+    // Called with each contact for our own processing. This is where we can
+    // test whether the given listeners are intersecting the trigger volume.
+    btScalar Trigger::addSingleResult(btManifoldPoint& cp,
+        const btCollisionObjectWrapper* colObj0, int partId0, int index0,
+        const btCollisionObjectWrapper* colObj1, int partId1, int index1) {
+        assert(colObj0->getCollisionObject() != trigger);
+        // The user pointer of the btRigidBody is our wrapping RigidBody instance.
+        auto body = static_cast<RigidBody*>(colObj0->getCollisionObject()->getUserPointer());
+        // Call the corresponding callback
+        observers[body](*body);
+        return 0; // Was a planned purpose, but is not used.
+    }
+
+    btCollisionObject* Trigger::GetCollisionObject() {
+        return trigger;
+    }
+
+    void Trigger::Process(btCollisionWorld& world) {
+        for (auto& observerPair : observers) {
+            world.contactPairTest(trigger, observerPair.first->GetRigidBody(), *this);
+        }
+    }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'OnEnter' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)